packages/contracts/contracts/core/SphinxGnosisSafeProxyFactory.sol

```diff
@@ -0,0 +1,166 @@
+
+        // Encode the call to the Gnosis Safe's `setup` function, which is an input to the
+        // Gnosis Safe proxy's `CREATE2` salt.
+        bytes memory safeInitializerData = abi.encodeWithSelector(
+            GnosisSafe.setup.selector,
+            _owners,
+            _threshold,
+            multiSend,
+            multiSendData,
+            _fallbackHandler,
+            // The following fields are for specifying an optional payment as part of the
+            // deployment. We don't use them.
+            address(0),
+            0,
+            address(0)
+        );
+
+        // Create the Gnosis Safe proxy's `CREATE2` salt. This salt ensures that the proxy's address
+        // changes if its initializer data changes. This salt is created in the exact same manner as
+        // the salt created in the Gnosis Safe Proxy Factory contract. The initializer data
+        // is hashed because it's cheaper than concatenating it.
+        bytes32 safeSalt = keccak256(abi.encodePacked(keccak256(safeInitializerData), _saltNonce));
+
+        // Deploy the Gnosis Safe proxy. Both versions of the proxy are functionally identical, but
+        // their bytecode is different because they have different metadata hashes. We make it
+        // possible to deploy either version to ensure that this contract is fully compatible with
+        // both Gnosis Safe v1.3.0 and v1.4.1.
+        safeProxy = _safeVersion == GnosisSafeVersion.v1_3_0
+            ? address(new GnosisSafeProxy_v1_3_0{ salt: safeSalt }(_safeSingleton))
+            : address(new GnosisSafeProxy_v1_4_1{ salt: safeSalt }(_safeSingleton));
+
+        emit DeployedGnosisSafeWithModule(
+            safeProxy,
+            moduleProxy,
+            _saltNonce,
+            _safeSingleton,
+            _safeVersion
+        );
+
+        // Initialize the Gnosis Safe proxy using the exact same data as the `safeInitializerData`
+        // above.
+        GnosisSafe(payable(safeProxy)).setup(
+            _owners,
+            _threshold,
+            address(multiSend),
+            multiSendData,
+            _fallbackHandler,
+            // The following fields are for specifying an optional payment as part of the
+            // deployment. We don't use them.
+            address(0),
+            0,
+            payable(address(0))
+        );
+
+        // Check that the Sphinx Module proxy is deployed and that it's enabled in the Gnosis Safe
+        // proxy. We don't need to check that the Gnosis Safe proxy is deployed because we deployed
+        // it via `CREATE2` above, which must have succeeded if we were able to make it to this
+        // point.
+        require(
+            moduleProxy.code.length > 0,
+            "SphinxGnosisSafeProxyFactory: module proxy not deployed"
+        );
+        require(
+            GnosisSafe(payable(safeProxy)).isModuleEnabled(moduleProxy),
+            "SphinxGnosisSafeProxyFactory: module proxy not enabled"
+        );
```

Comment:
These are the lines that prevent the griefing attack. An alternative approach is to check for the existence of contracts such as the Gnosis Safe singleton and the MultiSend contract, but I found that approach to be brittle because there's a lot of complexity in the Gnosis Safe proxy's initialization process, which means there's more surface area to trigger the griefing attack. I'm much more confident in this approach.